src/sentry/api/serializers/snuba.py

```diff
@@ -326,7 +332,10 @@ def serialize(self, result, column="count", order=None):
                 row.append(item)
             rv.append((k, row))
 
-        res = {"data": zerofill(rv, result.start, result.end, result.rollup)}
+        rv = zerofill(rv, result.start, result.end, result.rollup)
+        if self.max_buckets is not None and len(rv) > self.max_buckets:
+            rv = self.merge_buckets(rv)
+        res = {"data": rv}
```

Comment:
Is there going to be an issue with heterogenous or dynamic data based on parameter? If I am reading this properly (probably not) it looks like it will return min/max/avg only when max_buckets is exceeded otherwise it returns a single value? Or is it that the original result data already has these aggregates?

Reply:
Looking at the test data below it looks like it returns count if max_buckets isn't sent. What are your long-term plans for this API and do you have a transition plan?

Reply:
I discussed this with the frontend team and they said that they're happy enough to check for this data. Another option is returning a value in the response to indicate that aggregates have been returned, rather than raw values. I don't mind too much either way.